lib/trollop.rb

```diff
@@ -47,6 +244,56 @@ class VersionNeeded < StandardError
 ## and consider calling it from within
 ## Trollop::with_standard_exception_handling.
 class Parser
+
+  ## The registry is a class-variable Hash of registered option types.
+  ## By default, it contains flag(boolean), int, float, string, io and date types
+  ## and their plural versions.
+  @registry = {
+    # single opts
+    :fixnum => IntegerOption,
+    :int => IntegerOption,
+    :integer => IntegerOption,
+    :float => FloatOption,
+    :double => FloatOption,
+    :string => StringOption,
+    :bool => BooleanOption,
+    :boolean => BooleanOption,
+    :flag => BooleanOption,
+    :io => IOOption,
+    :date => DateOption,
+    :trueclass => BooleanOption,
+    :falseclass => BooleanOption,
+    ## array-arg opts
+    :fixnums => IntegerArrayOption,
+    :ints => IntegerArrayOption,
+    :integers => IntegerArrayOption,
+    :doubles => FloatArrayOption,
+    :floats => FloatArrayOption,
+    :strings => StringArrayOption,
+    :dates => DateArrayOption,
+    :ios => IOArrayOption,
+  }
+  
+  ## Gets the class from the registry.
+  ## Can be given either a class-name, e.g. Integer, a string, e.g "integer", or a symbol, e.g :integer
+  def self.registry_getklass(type)
+    return nil unless type
+    if type.respond_to?(:name)
+      type = type.name
+      lookup = type.downcase.to_sym
+    else
+      lookup = type.to_sym
+    end
+    raise ArgumentError, "Unsupported argument type '#{type}', registry lookup '#{lookup}'" unless @registry.has_key?(lookup)
+    return @registry[lookup]
+  end
+  
+  def self.registry_getopttype(type)
```

Comment:
The `nil` logic in here looks similar to `registry_getklass`

Does it make sense to just merge the methods?
We can move the `new` to that method. It already has a `key?`

Reply:
Agreed. Will merge the `registry_*` methods